version.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+# This script dynamically determines an appropriate version string for the
+# currently checked-out commit. As tag versions will typically be provided by CI
+# for releases, this script is primarily to support local builds that work as
+# one would expect.
```

Comment:
say I want to produce a local build with a specific version (e.g. "1.2.3"), how can I do that?

Reply:
```sh
export VERSION='v1.2.3'
make
```

Reply:
For what it's worth, if `HEAD` points to a tagged commit, `version.sh` will still append `+SHA` to the version string. If I recall correctly, I went back and forth on this, and decided that, ultimately, the chances of someone doing a local build and also being on a branch with tags at all is unlikely. Thinking about it some more, though, I can see how that might be more surprising behavior. I could definitely be convinced to return just the tag itself, but I'm curious if you have any thoughts there.